### PR TITLE
[Perf] Optimize SourceLocationConverter

### PR DESCRIPTION
**Pull request by rintaro**
* Implement `Sequence._copyContent(initializing:)` where needed, so `Array.append(contentsOf:)` is optimized
* In `RawSyntax.syntaxTextBytes`, copy consecutive `SyntaxText` at once, instead of copying for each token text.
* Small optimization in `SyntaxText.forEachLineLength(prefix:body:)`

```
Before:
average: 18.771, relative standard deviation: 0.979%, values: [19.052985, 19.032665, 18.911477, 18.644740, 18.631579, 18.601624, 18.566964, 18.623408, 18.683928, 18.957693]

After:
average: 9.473, relative standard deviation: 1.175%, values: [9.469722, 9.550222, 9.545583, 9.646544, 9.579049, 9.553190, 9.342481, 9.383785, 9.340003, 9.324244]
```

### CODE REVIEW

File: Sources/SwiftSyntax/SyntaxArenaAllocatedBuffer.swift [hunk 1]
```diff
@@ -61,22 +62,21 @@ public struct SyntaxArenaAllocatedBufferPointer<Element: Sendable>: RandomAccess
   /// - Important: The client needs to ensure sure that the buffer is indeed
   ///   allocated by a ``SyntaxArena`` and that the ``SyntaxArena`` will outlive
   ///   any users of this ``SyntaxArenaAllocatedBufferPointer``.
-  @_spi(RawSyntax) public init(_ buffer: UnsafeBufferPointer<Element>) {
+  public init(_ buffer: UnsafeBufferPointer<Element>) {
     self.buffer = buffer
   }
 
-  @_spi(RawSyntax)
```
Comment (rintaro) [inline Sources/SwiftSyntax/SyntaxArenaAllocatedBuffer.swift:68]: Removing `@_spi(RawSyntax)` is just a drive-by change. Since the type itself is already `@_spi(RawSyntax)`, the members don't need it.

File: Sources/SwiftSyntax/SourceLocation.swift [hunk 2]
```diff
@@ -522,38 +522,33 @@ fileprivate extension SyntaxText {
...
       /// * line-break → U+000A
       /// * line-break → U+000D
       /// * line-break → U+000D followed by U+000A
-      let isNewline = { () -> Bool in
-        if char == 10 { return true }
-        if char == 13 {
-          if curIdx < endIndex && self[curIdx] == 10 { advanceLengthByOne() }
-          return true
+      switch chr {
+      case 0x0A:
```
Comment (ahoppen) [inline Sources/SwiftSyntax/SourceLocation.swift:538]:
```suggestion
      case UInt8(ascii: "\n"):
```

Same below. I think that’s more explicit.

**Review by ahoppen — APPROVED** [review]

**Comment by rintaro** [comment]
@swift-ci Please test

File: Sources/SwiftSyntax/Syntax.swift [hunk 3]
```diff
@@ -365,20 +365,6 @@ extension Syntax: Identifiable {
...
-  ///
-  /// Unlike `description`, this provides a source-accurate representation
-  /// even in the presence of malformed UTF-8 in the input source.
-  ///
-  /// The ``SyntaxText`` arguments passed to the visitor are only guaranteed
-  /// to be valid within that call. It is unsafe to escape the `SyntaxValue`
-  /// values outside of the closure.
-  @_spi(RawSyntax)
-  public func withEachSyntaxText(body: (SyntaxText) throws -> Void) rethrows {
-    try raw.withEachSyntaxText(body: body)
```
Comment (rintaro) [inline Sources/SwiftSyntax/Syntax.swift:379]: This function was not used. It's not worth to expose this in `Syntax` it's a SPI anyway.

**Comment by rintaro** [comment]
@swift-ci Please test

**Comment by rintaro** [comment]
@swift-ci Please test Windows

File: Sources/SwiftSyntax/Raw/RawSyntax.swift [hunk 4]
```diff
@@ -361,16 +361,19 @@ extension RawSyntax {
 }
 
 extension RawTriviaPiece {
-  func withSyntaxText(body: (SyntaxText) throws -> Void) rethrows {
+  func withSyntaxText(body: (SyntaxText, _ isEphemeral: Bool) throws -> Void) rethrows {
```
Comment (ahoppen) [inline Sources/SwiftSyntax/Raw/RawSyntax.swift:364]: Could you add a comment what `isEmpemeral` means here?

**Review by ahoppen — APPROVED** [review]

**Comment by rintaro** [comment]
@swift-ci Please test

**Comment by rintaro** [comment]
@swift-ci Please test Windows

**Comment by rintaro** [comment]
@swift-ci Please test Windows
